File: zelixa-web/types/order.ts
export type OrderStatus = 'PENDING' | 'PAID' | 'PROCESSING' | 'DELIVERING' | 'COMPLETED' | 'CANCELLED' | 'COMPLAINT' | 'RETURNED' | 'all' | 'pending' | 'processing' | 'shipping' | 'completed' | 'cancelled';

export interface OrderItem {
  id: number | string;
  productId: number;
  productName?: string;
  name?: string;
  groupName?: string;
  size?: string;
  color?: string;
  quantity: number;
  qty?: number;
  price: number;
  img?: string;
  imageUrl?: string;
}

export interface Order {
  id: number | string;
  orderNumber?: string;
  userId?: number;
  totalAmount?: number;
  total?: number;
  status: OrderStatus;
  trackingNumber?: string;
  complaintNotes?: string;
  paymentToken?: string;
  paymentUrl?: string;
  shippingAmount?: number;
  shippingService?: string;
  createdAt?: string;
  updatedAt?: string;
  date?: string;
  items: OrderItem[];
  courier?: string;
  deliveryDate?: string;
}

export interface CheckoutRequest {
  shippingService?: string;
  destinationSubdistrictId?: string;
  shippingAmount?: number;
  voucherCode?: string;
  addressLine?: string;
  note?: string;
}

export interface CheckoutResponse extends Order {
  paymentToken: string;
  paymentUrl: string;
}

export interface OrderTab {
  id: OrderStatus | 'all';
  label: string;
}

File: zelixa-backend/src/main/java/com/zelixa/zelixa/entity/Order.java
package com.zelixa.zelixa.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "orders") // "order" is a reserved keyword in SQL
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "order_number", nullable = false, unique = true, length = 50)
    private String orderNumber;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(name = "total_amount", nullable = false)
    private Double totalAmount;

    @Column(nullable = false, length = 20)
    @Builder.Default
    private String status = "PENDING"; // PENDING, PAID, PROCESSING, DELIVERING, COMPLETED, CANCELLED, COMPLAINT,
                                       // RETURNED

    @Column(name = "tracking_number", length = 100)
    private String trackingNumber;

    @Column(name = "complaint_notes", length = 500)
    private String complaintNotes;

    @Column(name = "payment_token", length = 255)
    private String paymentToken;

    @Column(name = "payment_url", length = 500)
    private String paymentUrl;

    @Column(name = "shipping_amount")
    private Double shippingAmount;

    @Column(name = "shipping_service", length = 100)
    private String shippingService;

    @Column(name = "destination_subdistrict_id", length = 20)
    private String destinationSubdistrictId;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<OrderItem> items;
}
